Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -906,6 +906,13 @@ void Connection::processIncomingMessage(UniqueRef<Decoder> message)
     if (!m_syncState)
         return;
 
+    if (message->messageReceiverName() == ReceiverName::AsyncReply && (message->destinationID() & AsyncReplyHandlerWithProducerFlag)) {
```

Comment:
I did so to avoid looking into the map as an optimisation.

If the handler isn't found, it will default to the main thread code which will report the error and assert as needed.

I have now canned this version and instead added a `sendAsyncWithReplyOnDispatcher` which will take a RefCountedSerialFunctionDispatcher (so WorkQueue or RunLoop)


BTW, I think we can do `takeDirectDispatchAsyncReplyHandler` to not require two look-up, the take method will return a default constructed value, which will be null.

Reply:
I also hoped to avoid duplicating the map, but you're right, it does make things slightly cleaner that way